File: src/main/java/info/dddeurope/lab/app/views/SharingMap.java
package info.dddeurope.lab.app.views;

import java.util.HashMap;
import java.util.Map;

public class SharingMap {

    Map<String, String[]> postToUsers;
    
    public SharingMap() {
        this.postToUsers = new HashMap<>();
    }

    void userSharedPost(String postId, String userId) {
        //this.postToUsers.get(postId);
    }

}

File: DotNet/App/Views/SharingMap.cs
using System.Collections.Generic;

namespace App.Views
{
    public class SharingMap
    {
        Dictionary<string, string[]> _postToUsers;

        public SharingMap()
        {
            _postToUsers = new Dictionary<string, string[]>();
        }

        public void UserSharedPost(string postId, string userId)
        {
        }

        public string[] GetUsersThatSharedPost(string postId)
        {
            return new string[0];
        }
    }
}